web/public/js/init-master.js

```diff
@@ -754,6 +754,15 @@
           },
           templateUrl: '/directives/tpl/user/cd-profile/edit'
         })
+        //redirect to edit-user-profile page without need for userId in url
+        .state("edit-my-profile", {
+          url: '/profile/edit',
+          controller: function ($state, auth) {
+            auth.get_loggedin_user_promise().then(function(user) {
```

Comment:
What if not logged-in?
Maybe make it part of parent:dashboard too so that it'll redirect if that's not the case?